utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -105,6 +105,53 @@ function Invoke-ResourceRemoval {
             $null = $roleAssignmentsOnScope | Where-Object { $_.RoleAssignmentId -eq $ResourceId } | Remove-AzRoleAssignment
             break
         }
+        'Microsoft.Authorization/roleEligibilityScheduleRequests' {
+            $idElem = $ResourceId.Split('/')
+            $scope = $idElem[0..($idElem.Count - 5)] -join '/'
+            $pimRequestName = $idElem[-1]
+            $pimRoleAssignment = Get-AzRoleEligibilityScheduleRequest -Scope $scope -Name $pimRequestName
+            if ($pimRoleAssignment) {
+                $pimRoleAssignmentPrinicpalId = $pimRoleAssignment.PrincipalId
+                $pimRoleAssignmentRoleDefinitionId = $pimRoleAssignment.RoleDefinitionId
+                $guid = New-Guid
+                # PIM role assignments cannot be removed before 5 minutes from being created. Waiting for 5 minutes
+                Write-Verbose 'Waiting for 5 minutes before removing PIM role assignment' -Verbose
+                Start-Sleep -Seconds 300
+                $removalInputObject = @{
+                    Name             = $guid
+                    Scope            = $scope
+                    PrincipalId      = $pimRoleAssignmentPrinicpalId
+                    RequestType      = 'AdminRemove'
+                    RoleDefinitionId = $pimRoleAssignmentRoleDefinitionId
+                }
+                $null = New-AzRoleEligibilityScheduleRequest @removalInputObject
+
+            }
+            break
+        }
+        'Microsoft.Authorization/roleAssignmentScheduleRequests' {
```

Comment:
@eriqua, I guess we'd want to get this change in via a seperate PR?

Reply:
Agree

Reply:
Created a separate PR for this change: https://github.com/Azure/bicep-registry-modules/pull/4457